### ISSUE TITLE
Basis: Design and theme the Maintenance page

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#2528]
Current Basis
<img width="1434" alt="Screen Shot 2019-04-13 at 8 59 48 PM" src="https://user-images.githubusercontent.com/397895/56088124-19c00800-5e2f-11e9-951c-d92657335f61.png">

Bartik:
![backdrop-maintenance_mode-bartik](https://cloud.githubusercontent.com/assets/2423362/22839016/ba09d0c2-f01b-11e6-852c-3bb676ac298d.png)

Seven:
![backdrop-maintenance_mode-seven](https://cloud.githubusercontent.com/assets/2423362/22859769/4098e7a6-f13c-11e6-8d89-1c367462640a.png)

Previous Basis:
![backdrop-maintenance_mode-basis](https://cloud.githubusercontent.com/assets/2423362/22839026/c34496b8-f01b-11e6-8fea-da4322e67197.png)

---
PR: https://github.com/backdrop/backdrop/pull/2511
Milestone candidate tag for ~~1.12.3~~ meanwhile 1.12.7.

**Comment by Graham-72** [comment]
If we are in the process of loading a full update of Backdrop, is it reasonable to expect a fully-themed page here?

**Comment by klonos** [comment]
Nope, not a *fully* themed one - I was thinking at least just the basics like centering the text vertically and horizontally and keeping the fonts. Perhaps consider also rendering the header ribbon with the logo, site name and slogan. Basically, the minimum required for people to recognize the brand of the site.

Bartik does minimal things I believe:

https://github.com/backdrop/backdrop/blob/1.x/core/themes/bartik/template.php#L13
https://github.com/backdrop/backdrop/blob/1.x/core/themes/bartik/templates/maintenance-page.tpl.php
https://github.com/backdrop/backdrop/blob/1.x/core/themes/bartik/css/maintenance-page.css

Keep in mind that the intention here is to provide themers with something that they can easily override.

**Comment by klonos** [comment]
...the point is to be retaining a "professional" look when the site is in maintenance mode. As it is now, site visitors will most likely have the impression that the site is "broken" (instead of being in scheduled/intentional maintenance mode that is).

**Comment by klonos** [comment]
https://github.com/backdrop/backdrop/blob/1.x/core/themes/basis/template.php#L10

```php
/**
 * Implements hook_preprocess_page().
 *
 * @see maintenance_page.tpl.php
 */
function basis_preprocess_page(&$variables) {
```

The `@see` in the docblock in the main template file points to a maintenance page template that Basis does not include. Perhaps a copy-paste mistake or omission while @wesruv was coding Basis?

**Comment by klonos** [comment]
...even in maintenance mode, the login and password reset pages are themed, which is a good thing IMO. We just need to address a couple of issues: #2529

**Comment by klonos** [comment]
...updated issue summary to include a screenshot of the maintenance page for Seven (which is doing a great job BTW).

**Comment by klonos** [comment]
...I think I'm gonna take a shot at this a bit later, after I get some 💤

**Comment by jenlampton** [comment]
@klonos can you please create a separate issue for The @see in the docblock in the main template file?

From the screenshots it does look like the page is using the correct theme.  The correct fonts from Basis are appearing, for example. I think the problem is this...

> As it is now, site visitors will most likely have the impression that the site is "broken" (instead of being in scheduled/intentional maintenance mode that is).

That sounds like a design issue (and an important one!) but not really a bug. 

Also... Pushing this to the 1.6.2 milestone since there's no PR and we are releasing 1.6.1 tonight.

**Comment by jenlampton** [comment]
No PR here, removing milestone (but it can be added back as soon as we have a PR).

**Comment by stpaultim** [comment]
There was discussion today about how we might be creative with this page to improve the user experience process. How about displaying some literary quotes about dragons. It's cute and interesting content, even if you don't associate it with Backdrop.

https://www.thoughtco.com/writer-quotes-about-dragons-739550

**Comment by jenlampton** [comment]
no PR here yet, bumping milestone to 1.13.

**Comment by jenlampton** [comment]
stealing from @klonos

**Comment by jenlampton** [comment]
I've added a super minimal themed page in my [PR](https://github.com/backdrop/backdrop/pull/2511), like so: 
<img width="844" alt="screen shot 2019-02-14 at 5 32 27 pm" src="https://user-images.githubusercontent.com/397895/52828717-b90a8e00-307e-11e9-8464-7381c3bef106.png">

**Comment by klonos** [comment]
Tested the PR sandbox on various screen sizes and it works 👍 ...very minimal CSS changes, plus only limited to the maintenance page, so I don't mind getting it in as soon as we can. Having said that, leaving it up to @quicksketch to have the final word.

**Comment by klonos** [comment]
...making a note to revisit #2529 once this is merged.

**Comment by herbdool** [comment]
I think 1.12.3 makes sense.

**Comment by stpaultim** [comment]
During a call back in December there was some discussion about doing something creative to the maintenance mode page to improve the user experience and have a little fun with Backdrop. The current PR is a clear improvement and meets the original requirements outlined in this issue, but I don't think it quite qualifies in addressing the "creativity" criteria (sorry @jenlampton). 

I don't want to slow this PR down. Should I open a new issue to "Make the maintenance mode page delightful"? Or is there still time and interest in including additional work in this PR? Or have we decided that we want to keep Backdrop dull and boring ;-)

I suggested randomly displaying a literary quote about dragons on the page. I can also imagine a drawing of Drop the dragon carrying a picket sign that says "Site Under Maintenance." @wesruv suggested that maybe we have an image of Drop the dragon with a shovel playing homage to this classic image:

![construction](https://user-images.githubusercontent.com/3144571/52918508-e85f0c00-32c5-11e9-98c0-e18758f3ab6b.png)

Do we have someone with the skills to create some drawings of Drop the Dragon in various poses that we can use in other places?

**Comment by herbdool** [comment]
I lean towards getting this small change in the next bug release and have a new issue for being more creative with dragons. That way the new issue can percolate a bit with creativity and we can still avoid the ugly current maintenance page.

**Comment by Radcliffe** [comment]
Google Chrome has a hidden dinosaur game that appear when the user is not connected to the Internet. It would be cute to have a simple dragon-themed game on the maintenance page to amuse people while they wait for the site to come back online. Probably too cute.

**Comment by klonos** [comment]
> I lean towards getting this small change in the next bug release and have a new issue for being more creative with dragons. That way the new issue can percolate a bit with creativity and we can still avoid the ugly current maintenance page.

That ^^ 👍

**Comment by quicksketch** [comment]
Per our own directions for CSS in Basis, this should be a component I think instead of being in skin.css:

```
The types of CSS files in the theme include:
* Base      - Normalizes rendering across browsers.
* Layout    - Modifies page layout.
* Component - Specific styles for individual pieces of functionality.
* Skin      - The colors, fonts, and aesthetic CSS
```

Though it seems silly to add this CSS to every page of the site. Perhaps we should add a `basis_preprocess_maintenance_page()` to add that CSS file only on the maintenance page?

**Comment by jenlampton** [comment]
good ideas, all done in the latest PR.

**Comment by klonos** [comment]
LGTM 👍

**Comment by olafgrabienski** [comment]
From https://github.com/backdrop/backdrop/pull/2511#pullrequestreview-204125639:

> Basis uses 'rem' instead of 'px' for padding, so far without exceptions, would be good to stick with it.

Hm, looks like the PR uses still `px` instead of `rem`. @jenlampton had you seen the review?

**Comment by klonos** [comment]
...back to NW in order to convert `px`s to `em`s/`rem`s.

**Comment by olafgrabienski** [comment]
I've had a look at the sandbox site, the maintenance page looks good to me!

(And `px` were converted to `rem` in the PR a while ago.)

**Comment by jenlampton** [comment]
Adding the RTBC label for core committer review (and merge?)

**Comment by quicksketch** [comment]
Thanks folks! Merged https://github.com/backdrop/backdrop/pull/2511 into 1.x and 1.12.x.
